sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/README.md

```diff
@@ -1,13 +1,227 @@
+The priority of which value takes on the instrumentation key is:
+* Key from explicitly passed in connection string
+* Key from explicitly passed in instrumentation key
+* Key from connection string in environment variable
+* Key from instrumentation key in environment variable
+Please find the samples linked below for demonstration as to how to authenticate via either approach.
+
+#### [Create Exporter from connection string][sample_authenticate_client_connstr]
+
+```Python
+from microsoft.opentelemetry.exporter.azuremonitor import AzureMonitorSpanExporter
+exporter = AzureMonitorSpanExporter(
+    connection_string = os.environ["AZURE_MONITOR_CONNECTION_STRING"]
+)
+```
+
+#### Create Exporter using the instrumentation key:
+
+```Python
+from microsoft.opentelemetry.exporter.azuremonitor import AzureMonitorSpanExporter
+exporter = AzureMonitorSpanExporter(
+    instrumentation_key = os.environ["AZURE_MONITOR_INSTRUMENTATION_KEY"]
+)
+```
+
 ## Key concepts
 
+Some of the key concepts for the Azure monitor exporter include:
+
+* [Opentelemetry][opentelemtry_spec]: Opentelemetry is a tool to instrument, generate, collect, and export telemetry data (metrics, logs, and traces) for analysis in order to understand your software's performance and behavior.
+
+* [Trace][trace_concept]: Trace can be thought of as a directed acyclic graph (DAG) of Spans, where the edges between Spans are defined as parent/child relationship.
+
+* [AzureMonitorSpanExporter][client_reference]: This is the object a user should first initialize to connect the namespace to send trace data.
+
+* [Exporter Options][exporter_options]: Options to configure Azure exporters. Includes connection_string, instrumentation_key, proxies, timeout etc.
+
+For more information about these resources, see [What is Azure Monitor?][product_docs].
+
 ## Examples
 
+The following sections provide several code snippets covering some of the most common tasks, including:
+
+* [Export hello world trace data](#export-hello-world-trace)
+* [Modifying Traces](#modifying-traces)
+* [Instrumentation with requests library](#instrumentation-with-requests-library)
+
+
+### Export Hello World Trace
+
+```Python
+import os
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+from microsoft.opentelemetry.exporter.azuremonitor import AzureMonitorSpanExporter
+
+# Callback function to add os_type: linux to span properties
+def callback_function(envelope):
+    envelope.data.base_data.properties["os_type"] = "linux"
+    return True
+
+exporter = AzureMonitorSpanExporter(
+    connection_string = os.environ["AZURE_MONITOR_CONNECTION_STRING"]
+)
+exporter.add_telemetry_processor(callback_function)
+
+trace.set_tracer_provider(TracerProvider())
```

Comment:
What about tracer provider, span processor and sampler? These are terms user probably should be familiar with. If not in our docs here, we should at least link to the Opentelemetry docs explaining what they are.

Reply:
will add them in key concepts